onnxruntime/core/session/inference_session.cc

```diff
@@ -1395,6 +1419,42 @@ common::Status InferenceSession::Initialize() {
       // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
       ORT_RETURN_IF_ERROR_SESSIONID_(graph.Resolve());
 
+      // Currently only the CUDA EP is considered.
+      // If the CUDA EP is part of the providers list for this session AND
+      // The CUDA EP is configured to do a graph capture AND
+      // All the graph nodes have been assigned to the CUDA EP,
+      // Then the CUDA EP is cached for triggering a ReplayGraph() in Run().
+      for (auto execution_provider : execution_providers_) {
```

Comment:
We don't need to check for all execution providers here. Can just check for the existence of CUDA.

Reply:
I thought this kind of hard code needs to be avoided here. Sure, I changed it back to be as below:
```
auto* cuda_ep = execution_providers_.Get(onnxruntime::kCudaExecutionProvider);
if (cuda_ep && cuda_ep->IsGraphCaptureEnabled()) {
  ...     
```